### PR TITLE
webui: extend restore to handle plugin options

### PR DESCRIPTION
**Pull request by fbergkemper**
Extends the restore module to allow the input of plugin options for restoring backup jobs that used a plugin.

#### Please check

- [x] Short description and the purpose of this PR is present _above this paragraph_
- [x] Your name is present in the AUTHORS file (optional)

If you have any questions or problems, please give a comment in the PR.

### Helpful documentation and best practices

- [Git Workflow](https://docs.bareos.org/DeveloperGuide/gitworkflow.html)
- [Automatic Sourcecode Formatting](https://docs.bareos.org/DeveloperGuide/generaldevel.html#automatic-sourcecode-formatting)
- [Check your commit messages](https://docs.bareos.org/DeveloperGuide/gitworkflow.html#commits)

##### General
- [x] PR name is meaningful
- [x] Purpose of the PR is understood
- [x] Commit descriptions are understandable and well formatted
- [x] Is the PR title usable as CHANGELOG entry?
- [x] Separate commit for CHANGELOG.md ("update CHANGELOG.md"). The PR number is correct.

##### Source code quality

- [x] Source code changes are understandable
- [x] Variable and function names are meaningful
- [x] Code comments are correct (logically and spelling)
- [x] Required documentation changes are present and part of the PR
- [x] `bareos-check-sources --since-merge` does not report any problems

### CODE REVIEW

File: webui/module/Restore/src/Restore/Model/Restore.php [hunk 1]
```diff
@@ -184,6 +177,25 @@ public function getInputFilter()
             )
          ));
 
+         $inputFilter->add(array(
+            'name' => 'pluginoptions',
+            'required' => false,
+            'filters' => array(
+               //array('name' => 'StripTags'),
+               //array('name' => 'StringTrim'),
```
[the remark this answers is not in the document]
Comment (fbergkemper) [reply webui/module/Restore/src/Restore/Model/Restore.php]: Applied, see https://github.com/bareos/bareos/pull/1276/commits/fb25cb0e2c06e3b228ae734417e1f249527b04ad .

File: webui/module/Restore/src/Restore/Form/RestoreForm.php [hunk 2]
```diff
@@ -411,6 +411,83 @@ public function __construct($restore_params=null, $clients=null, $filesets=null,
...
+                     'type' => 'text',
+                     'options' => array(
+                        'label' => _('Plugin Options')
+                        ),
+                     'attributes' => array(
+                        'class' => 'form-control selectpicker show-tick',
+                        'value' => '',
+                        'id' => 'pluginoptions',
+                        'size' => '30',
+                        'placeholder' => _('e.g. <plugin>:file=<filepath>:reader=<readprogram>:writer=<writeprogram>'),
```
[the remark this answers is not in the document]
Comment (fbergkemper) [reply webui/module/Restore/src/Restore/Form/RestoreForm.php]: Applied, see https://github.com/bareos/bareos/pull/1276/commits/c1e83d30523f59cf0e5d0948d2c9a08cfdc66781 .

File: webui/module/Client/src/Client/Model/ClientModel.php [hunk 3]
```diff
@@ -114,6 +114,24 @@ public function getClientBackups(&$bsock=null, $client=null, $fileset=null, $ord
...
          $backups = \Zend\Json\Json::decode($result, \Zend\Json\Json::TYPE_ARRAY);
+
+         if(!isset($limit)) {
+            $filesets_pluginjob = array();
+            foreach ($backups['result']['backups'] as $key=>$backup) {
+               if (!array_key_exists($backup['fileset'], $filesets_pluginjob)) {
+                  $cmd = 'show fileset="'.$backup['fileset'].'"';
+                  $result = $bsock->send_command($cmd, 2);
+                  $fileset = \Zend\Json\Json::decode($result, \Zend\Json\Json::TYPE_ARRAY);
+                  $fileset_pluginjob[$backup['fileset']] = false;
```
Comment (sduehr) [inline webui/module/Client/src/Client/Model/ClientModel.php:125]:
```suggestion
                  $filesets_pluginjob[$backup['fileset']] = false;
```

File: webui/module/Client/src/Client/Model/ClientModel.php [hunk 4]
```diff
@@ -114,6 +114,24 @@ public function getClientBackups(&$bsock=null, $client=null, $fileset=null, $ord
...
+                  if (array_key_exists("plugin", $fileset['result']['filesets'][$backup['fileset']]['include'][0])) {
+                     if (!empty($fileset['result']['filesets'][$backup['fileset']]['include'][0]['plugin'])) {
+                        $fileset_pluginjob[$backup['fileset']] = true;
```
Comment (sduehr) [inline webui/module/Client/src/Client/Model/ClientModel.php:128]:
```suggestion
                        $filesets_pluginjob[$backup['fileset']] = true;
```

File: webui/module/Client/src/Client/Model/ClientModel.php [hunk 5]
```diff
@@ -114,6 +114,24 @@ public function getClientBackups(&$bsock=null, $client=null, $fileset=null, $ord
...
+                     }
+                  }
+               }
+               $backups['result']['backups'][$key]['pluginjob'] = $fileset_pluginjob[$backup['fileset']];
```
Comment (sduehr) [inline webui/module/Client/src/Client/Model/ClientModel.php:132]:
```suggestion
               $backups['result']['backups'][$key]['pluginjob'] = $filesets_pluginjob[$backup['fileset']];
```

**Review by sduehr — CHANGES_REQUESTED** [review]
Or remove the "s" in $filesets_pluginjob in the lines 119 and 121 of ClientModel.php

File: webui/module/Client/src/Client/Model/ClientModel.php (hunk 5 above)
Comment (fbergkemper) [reply webui/module/Client/src/Client/Model/ClientModel.php]: Typo fixed in https://github.com/bareos/bareos/pull/1276/commits/f0cf31ae7030b95a0789d8d1f8a0d2c22188d22a .

**Review by sduehr — APPROVED** [review]
Thanks, looks good now.

**Pull request merged**